From: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Subject: calibrate: update header inclusion
Date: Tue, 25 Nov 2025 00:06:06 +0100

While cleaning up some headers, I got a build error on this file:

init/calibrate.c:20:9: error: call to undeclared function 'kstrtoul'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]

Update header inclusions to follow IWYU (Include What You Use) principle.

Link: https://lkml.kernel.org/r/20251124230607.1445421-1-andriy.shevchenko@linux.intel.com
Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 init/calibrate.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/init/calibrate.c~calibrate-update-header-inclusion
+++ a/init/calibrate.c
@@ -5,12 +5,15 @@
  *  Copyright (C) 1991, 1992  Linus Torvalds
  */
 
-#include <linux/jiffies.h>
 #include <linux/delay.h>
 #include <linux/init.h>
-#include <linux/timex.h>
-#include <linux/smp.h>
+#include <linux/jiffies.h>
+#include <linux/kstrtox.h>
 #include <linux/percpu.h>
+#include <linux/printk.h>
+#include <linux/smp.h>
+#include <linux/stddef.h>
+#include <linux/timex.h>
 
 unsigned long lpj_fine;
 unsigned long preset_lpj;
_
